src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5879,7 +5907,7 @@ private bool IsUsingAliasInScope(string name)
 
         private void WarnOnAccessOfOffDefault(SyntaxNode node, BoundExpression boundLeft, DiagnosticBag diagnostics)
         {
-            if (boundLeft != null && boundLeft.Kind == BoundKind.DefaultExpression && boundLeft.ConstantValue == ConstantValue.Null &&
+            if (boundLeft != null && boundLeft is BoundDefaultExpression && boundLeft.ConstantValue == ConstantValue.Null &&
```

Comment:
>if (boundLeft != null && boundLeft is BoundDefaultExpression && boundLeft.ConstantValue == ConstantValue.Null && [](start = 12, length = 112)

It looks like we are going to stop reporting WRN_DotOnDefault for ```default(C).ToString()```. Is this intentional and why?

Reply:
`BoundConvertedDefaultExpression` derives from `BoundDefaultExpression` so the warning is reported. I'll add a test.
That said, the language version check seems wrong (I'll remove and document).

---
In reply to: [314087501](https://github.com/dotnet/roslyn/pull/37596#discussion_r314087501) [](ancestors = 314087501)

Reply:
> the language version check seems wrong 

I think the intent was to replace it with corresponding nullable warning and have it be enabled/disabled as all other nullable warnings.

---
In reply to: [314092390](https://github.com/dotnet/roslyn/pull/37596#discussion_r314092390) [](ancestors = 314092390,314087501)